js/spec/index-spec.js

```diff
@@ -562,6 +562,35 @@ describe('Node Sentinel File Watcher', function() {
         watch = null;
       }
     });
+
+    it('Can pass falsy values to errorCallback', async function() {
+      const ok = [undefined, null, 0, '', false];
+      const notOk = [1, true, 'a', {}, []];
+      await Promise.all([
+        Promise.all(ok.map(
+          errorCallback => nsfw(
+            workDir,
+            () => {},
+            { errorCallback }
+          )
+        )),
+        Promise.all(notOk.map(
+          async errorCallback => {
+            try {
+              await nsfw(
+                workDir,
+                () => {},
+                { errorCallback },
+              );
+            } catch (error) {
+              assert.ok(error.message === 'options.errorCallback must be a function.');
+              return;
+            }
+            assert.fail('nsfw must error if errorCallback is not a function nor a falsy value');
+          }
+        ))
+      ]);
+    });
```

Comment:
Something about this unit test is causing the garbage collection test to fail, That's pretty spooky, but it likely means nsfw holds onto these function references. Best move forward would be to skip this test for now, write up an issue for tracking down persistent references on init without calling `stop`, and then solve that in a separate issue where we turn this test back on as evidence that we've addressed the GC issues.